### DISCUSSION TITLE
ChatGPT and Whisper APIs released. Can this now be implemented?

### DISCUSSION THREAD

**Discussion by emmanuelnk** [TheR1D/shell_gpt#48] [Ideas]
See this: https://openai.com/blog/introducing-chatgpt-and-whisper-apis

> ChatGPT and Whisper models are now available on our API, giving developers access to cutting-edge language (not just chat!) and speech-to-text capabilities. Through a series of system-wide optimizations, we’ve achieved 90% cost reduction for ChatGPT since December; we’re now passing through those savings to API users. Developers can now use our open-source Whisper large-v2 model in the API with much faster and cost-effective results. ChatGPT API users can expect continuous model improvements and the option to choose dedicated capacity for deeper control over the models. We’ve also listened closely to feedback from our developers and refined our API terms of service to better meet their needs.

What would it take to implement ChatGPT API? Any help needed?

**Comment by TheR1D** [comment]
Thank you for bringing this up.

It is already implemented in PR https://github.com/TheR1D/shell_gpt/pull/47
And [released](https://github.com/TheR1D/shell_gpt/releases) as v0.6.0
You can install latest version using pip: `pip install shell-gpt --upgrade --user`

**Comment by edshamis** [comment]
@TheR1D any plans to add Whisper support?

**Reply by TheR1D** [reply]
No, ShellGPT is a programming and system administration tool, while speech recognition is a different technology altogether.
